# Tute notes

In [1]:
import math

# Problem 9.1

In [2]:
d = 1
r_eye = 0.015
A_pupil = 0.0001
c = 2.99792458E+8

In [3]:
L = 100 / (4 * math.pi * d**2)
L

7.957747154594767

In [4]:
((2*r_eye) * A_pupil * L) / (4 * math.pi * d**2 * c)

6.336957927386663e-15

# 10.3

If sun were 100% hydrogen, it would have N atoms:

$$N = \frac{M_{\odot}}{m_H} = \frac{ 1.9891*10^{30}kg }{ 1.6735575 * 10^{-27}kg } = 1.189*10^{57}kg$$



In [5]:
M_sun = 1.9891E+30
L_sun = 3.839E+26
E_ev = 10
E = 1.60218e-18
m_H = 1.6735575E-27
N = M_sun / m_H
N

1.1885459567418508e+57

If each atom has energy $E$ = 10eV, time would be:

$$
t = \frac{NE}{L_{\odot}} = 4.96E * 10^{12}s = 1.57*10^5 yr
$$

In [6]:
t_s = (N * E) / L_sun
t_y = t_s / 60 / 60 / 24 / 365.25
t_s, t_y

(4960314042648.238, 157182.8669685983)

# 10.10

Calculate the ratio of the energy generation rate for the `pp` chain to the energy generation rate for the `CNO` cycle given conditions characteristic of the center of the present-day (evolved) Sun, namely:

$$T = 1.5696 * 10^7 K$$
$$\rho = 1.527 x 10^5 kg m-3$$
$$X = 0.3397$$ 
$$X_{CNO} = 0.0141$$

Assume that the pp chain screening factor is unity ($f_{pp} = 1$) and that the pp chain branching
factor is unity ($\Psi_{pp} = 1$).

i.e. Calculate and compare energy generation rate for pp and CNO.

### PP

Simplified version written as ‘power law’ for cases where $T \approx 1.5*10^7K$:

$\epsilon_{pp} \approx \epsilon_{0,pp}^{\prime} \rho X^2 f_{pp} \psi_{pp}C_{pp}T_6^{4}$ W kg-1

Where:

- $\epsilon_{0,pp}^{\prime} = 1.08*10^{-12}$ W m3 kg-2
- $T_6$ is dimensionless temp in units $10^6K$ ($T_6 = T/10^6K$)
- $f_{pp} = f_{pp}(X, Y, \rho, T)$ is the pp chain screening factor
- $\psi_{pp} = \psi_{pp}(X, Y, T)$ is a factor to correct for simultaneous PP1, PP2 and PP3
- $C_{pp} \sim 1$ is “higher order correction of terms”



In [7]:
e_0pp = 1.08E-12
T = 1.5696E+7
T_6 = T / 1.0E+6
rho = 1.527E+5
X = 0.3397
X_CNO = 0.0141
f_pp = 1
psi_pp = 1
C_pp = 1
C_CNO = 1 # ASSUMPTION! - NOT GIVEN IN QUESTION.

In [8]:
# PP
e_pp = e_0pp * rho * (X**2) * f_pp * psi_pp * C_pp * (T_6**(4))
e_pp

0.001155074101498429

### CNO

$\epsilon_{CNO} = 8.67*10^{20}\rho XX_{CNO}C_{CNO}T_6^{-2/3}e^{-152.28T_6^{-1/3}}$ W kg-1

## CNO

Simplified version written as ‘power law’ for cases where $T \approx 1.5*10^7K$:

$\epsilon_{CNO} \approx \epsilon_{0,CNO}^{\prime} \rho XX_{CNO} T_6^{19.9}$ W kg-1

Where:

- $\epsilon_{0,CNO}^{\prime} = 8.24*10^{-31}$ W m3 kg-2


In [9]:
e_0CNO = 8.24E-31

e_CNO = e_0CNO * rho * X * X_CNO * (T_6**19.9)
e_CNO

0.0003769483181974039

In [10]:
ratio = e_pp / e_CNO
print('pp chain generates %s times more energy than CNO cycle' % ratio)

pp chain generates 3.064277105737155 times more energy than CNO cycle


# 10.14

$^{235}_{92}U \rightarrow ^{4}_{2}He + ^{231}_{90}Th$

In [11]:
p1 = 92
p2 = 90 + 2

n1 = 235
n2 = 4 + 231
p1 == p2, n1 == n2

(True, True)

In [12]:
27-14

13

## PP1 10.37

Leptons?

2x Hydrogen atoms combining.
They have: 

- 1 proton each
- 0 neutrons each (mass number = 1 = Z)

They create:

- Helium - 2x protons
- 0 neutrons (mass number = 2 = Z)
- positron
- neutrino

#### Also 10.48

Where photon comes from? \o/

- Electron must also have a companion anti-neutrino, or a positron on opposing side of equation
- Positron must also have a companion neutrino, or an electron on opposing side of equation
- Photons are by themselves.




# TODO

- write down simplified version of energy generation e.g. `10.59`
- triple alpha process 10.60 etc..
- mostly focused on nucleosynthesis
- StatStar
   - initial conditions must be just right, otherwise error
    - 


# 11.2

(a) At what rate is the Sun's mass decreasing due to nuclear reactions? Express your answer in solar masses per year.


$$E = mc^2$$

$$t = 1 * 365.25 * 24 * 60 * 60 = 31557600s$$

$$t = \frac{E_{nuclear}}{L_{\odot}}$$

$$E_{nuclear} = tL_{\odot}$$ 

$$E_{nuclear} = 31557600 * 3.839*10^{26} = 1.21 * 10^{34}$$

$$\frac{E_{nuclear}}{c^2} = m$$

$$\frac{1.21 * 10^{34}}{c^2} = 1.34*10^{17}kg = 6.77*10^{-14}M_{\odot}$$

In [13]:
E = 31557600 * L_sun
E

1.2114962639999998e+34

In [14]:
m = E/(c**2)
m

1.3479713860483491e+17

In [15]:
(m / M_sun) * 10**9

6.776790438129552e-05

# 11.5

### (a) Using Eq. (9.63) and neglecting turbulence, estimate the full width at half-maximum of the hydrogen $H\alpha$ absorption line due to random thermal motions in the Sun's photosphere.

Assume that the temperature is the Sun's effective temperature.

Eq 9.63, p269

In [16]:
# wavelgnth for H_alpha is
lam = 656.3E-6
T = 5777
k = 1.3806504E-23
m = 1.6735575E-27

In [17]:
delta_lambda_1_2 = ((2*lam)/c) * math.sqrt( ((2*k*T)/m) * math.log(2) )
delta_lambda_1_2

3.558870223299679e-08

### (b) Using $H\alpha$ redshift data for solar granulation, estimate the full width at half-maximum when convective turbulent motions are included with thermal motions.

In [18]:
# from p364
v_turb = 400

delta_lambda_1_2_turb = ((2*lam)/c) * math.sqrt( ( (((2*k*T)/m) + (v_turb**2))) * math.log(2) )
delta_lambda_1_2_turb 

3.5618559122072825e-08

### (c) What is the ratio of $v^2_{turb}$ to $2kT/m$?

In [19]:
(v_turb**2) / ((2*k*T)/m)

0.0016785896261711658

### (d) Determine the relative change in the full width at half-maximum due to Doppler broadening when turbulence is included. Does turbulence make a significant contribution to $\lambda_{1/2}$ in the solar photosphere?

#### Answer
The relative change is $0.083$%, which is not a significant contribution.

In [ ]:
(delta_lambda_1_2_turb - delta_lambda_1_2) / delta_lambda_1_2_turb * 100

# 11.7

### (a) Using Eq. (3.22), show that in the Sun's photosphere,

$$ln(B_a/B_b) \approx 11.5 + \frac{hc}{kT}(\frac{1}{\lambda_b} - \frac{1}{\lambda_a})$$


where $B_a/B_b$, is the ratio of the amount of blackbody radiation emitted at $\lambda_a = 10nm$ to
the amount emitted at $\lambda_b = 100nm$, centered in a wavelength band 0.1 nm wide.

Eq3.22 is Plank function:

$$B_{\lambda}(T) = \frac{2hc^2 / \lambda^5}{e^{hc / \lambda kT} -1}$$

Where

$h$ = planks constant = $6.62606876 * 10^{-34} J s$

---

#### Answer (incorrect - you need to integrate over $\Delta \lambda$

We'll assume temp for _surface_ of the photosphere of $T=4400K$

First, we need to find $B_a, B_b$ using plank function.


$$B_{a}(T) = 4.21 *10^9$$

$$B_{b}(T) = 4.72 *10^5$$

In [20]:
lambda_a = 10.0E-9 # 10nm
lambda_b = 100.0E-9 # 100nm
T = 4400
h = 6.62606876E-34
c = 299792458
k = 1.3806503E-23   

def plank(lambd, T): 
    return ( 
        ((2 * h * c**2) / lambd**5) / 
        (math.e**((h*c) / (lambd * k * T)) - 1)
    )
        
B_a = plank(lambda_a, T)
B_b = plank(lambda_b, T)
        
B_a, B_b       

(1.1589842422445282e-118, 74944.94737577756)

Now we can evaluate each side

In [21]:
LHS = math.log(B_a/B_b)
LHS

-282.7820060930181

In [22]:
RHS = 11.5 + ((h*c)/(k*T))*((1/lambda_b)-(1/lambda_a))
RHS

-282.7949315579883

### (b) What is the value of this expression for the case where the temperature is taken to be the effective temperature of the Sun?

In [23]:
T = 5777
B_a = plank(lambda_a, T)
B_b = plank(lambda_b, T)

RHS = 11.5 + ( (h*c)/(k*T) ) * ( (1/lambda_b) - (1/lambda_a) )

LHS = math.log(B_a/B_b)
LHS, RHS

(-212.6341575980792, -212.64708306303422)

### (c) Writing the ratio in the form $B_a/B_b = 10^x$, determine the value of x.

In [24]:
math.log((B_a/B_b), 10)

-92.3458413089922

# 11.8

The gas pressure at the base of the photosphere is approximately $2 * 10^4$ N m-2 and the mass density is $3.2 * 10^{-4}$ kg m-3
 
Estimate the sound speed at the base of the photosphere, and compare your answer with the values at the top of the photosphere and averaged throughout the Sun.

---

Adiabatic Sound Speed: 

$$v_s = \sqrt{\gamma P / \rho} = 10206 m/s = 10.2km/s$$

Assume monotomic gas ($\gamma = 5/3$).

In [25]:
P = 2.0E+4
rho = 3.2E-4
gamma = 5/3

v_base = math.sqrt( (gamma*P) / rho)
v_base

10206.207261596575

$v_{s,top} = 6.9km/s$ (Example 11.2.2, p379)

$v_{s,sun} = 400km/s$ (Example 10.4.2)

Speed of sound increases as we go deeper into the sun - pressure increases faster than density.

# 11.14

Assuming that an average of one coronal mass ejection occurs per day and that a typical CME ejects $10^{13}$ kg of material, estimate the annual mass loss from CMEs and compare your answer with the annual mass loss from the solar wind. 

Express your answer as a percentage of CME mass loss to solar wind mass loss.

In [26]:
wind_yr = 3.0E-14 # from p374
cme_day = 10**13
cme_yr = (cme_day * 325.25) / M_sun

percent =  (cme_yr / wind_yr) * 100

cme_yr, wind_yr, percent

(1.6351616308883416e-15, 3e-14, 5.450538769627806)

# 12.3 
The Boltzmann factor $e^{-(E_2-E_1)/kT}$, helps determine the relative populations of energy levels (see Section 8.1). 

Using the Boltzmann factor, estimate the temperature required for a hydrogen atom's electron and proton to go from being anti-aligned to being aligned. 

Are the temperatures in H I clouds sufficient to produce this low-energy excited state?

### Answer

Ratio of number of atoms in two states (p212):

$$\frac{N_b}{N_a} = \frac{g_b}{g_a} e^{-(E_b-E_a)/kT}$$

$k$ = Boltzmann constant
$E_a,E_b$ are energies (eV)
$g_a,g_b$ are number of degenerate states (i.e. how many different quantum states are there which have the same energy) , e.g.  for hydrogen:

$$g_b = 2n^2$$

From p212, we see that $n=1$ for both ground states, so:

$$
g_a, g_b = 2 * 1^2 = 2
$$

So the equation can be simplified $(2/2 = 1)$:

$$
1 = e^{-(E_2-E_1)/kT}
$$

Ground state:
$E_1 = -13.6eV$

Excited state (difference is the energy of the photon, right?):

$\lambda = 0.211m$

$E_2 = \frac{hc}{\lambda} = 5.876 * 10^{-6} eV$


i.e. the energy change is negligible, $E$ still is essentially still $=E_1$

From Example 8.1.3, p 213:

$$
T = \frac{E}{k} = \frac{5.876 * 10^{-6} eV}{8.617343 * 10^{-5}eV K^{-1} \ln(4)} = 0.068 K
$$

**Temperatures in diffuse H I clouds ARE sufficient (only 30-80K).**

In [27]:
λ = 0.211
h_ev = 4.13566733e-15 # eV
ev_factor = 1.60218e-19
c = 299792459

k = 1.3806503e-23
k_ev = 8.617343E-5

E_1 = -13.6 
E_2 = ((h_ev * c) / λ)
E = E_2
E

5.876027860031586e-06

In [28]:
E_2 / k_ev

0.0681883947294611

# 12.5
Using an approach analogous to the development of Eq. (10.29) for nuclear reaction rates, make a crude `estimate of the number of random collisions` per cubic meter per second between CO and H, molecules in a giant molecular cloud that has a temperature of `15K` and a number
density of $n_{H_2} = 10^8 m^{-3}$. 

Assume (incorrectly) that the molecules are spherical in shape with radii of approximately `0.1nm`, the characteristic size of an atom.

### Answer

$T = 15K$

$n_{H_2} = 10^8 m^{-3}$

$m = 10^5 M_{\odot}$

$r = 0.1nm$

$r_{gmc} = 50/2pc$

$v_{gmc} = 4\pi r_{gmc}^2 = 7.48 * 10^{36} m^3$

In [30]:
T = 15
n_H2 = 10.0E+8
m = 5 * M_sun
r = 0.1E-9

r_gmc = (50/2) * 3.086e+16 # to meters
v_gmc = 4 * math.pi * r_gmc**2

v_gmc

7.479657727706605e+36

# 12.8
(a) Equations (12.10) and (12.11) illustrate a cooling mechanism for a molecular cloud accomplished through the excitation of oxygen atoms. 

Explain why the excitation of hydrogen rather than oxygen is not an effective cooling mechanism.

### Answer

p411

12.10 - O and H collide with kinetic energy, which puts O in excited state, denoted by $O^*$:
$$
O + H \rightarrow O^* + H
$$

12.11 - Excited $O^*$ decays to ground state O, emitting an IR photon of suffient wavelength to escape the cloud (Mie scattering), therefore removing energy/temp from the cloud):
$$
O^* \rightarrow O + \gamma
$$

H is much more abundant than O. If H emits a photon, it is much more likely to be reabsobed by a H (i.e. less likely to escape the cloud).

(b) Why are the temperatures of hot cores significantly greater than dense cores?

### Answer

Note - "hot cores", and "dense cores" are a types of interstellar cloud (p408).

Hot cores are much more dense than dense cores (despite the name...).
i.e. in the more dense "hot core" it's harder for photons to espace, hot cores also contain large O and B stars.

# 12.14
Assuming that the free-fall acceleration of the surface of a collapsing cloud remains constant during the entire collapse, derive an expression for the free-fall time. 

Show that your answer differs from Eq. (12.26) only by a term of order unity.

---

# Main Sequence & Post-Main-Sequence Stellar Evolution

# 13.1

(a) For a 5$M_{\odot}$ star, use the data in Table 13.1 associated with Fig. 13.1 to construct a table
that expresses the evolutionary times between points 2 and 3, points 3 and 4, and so on,
as a percentage of the lifetime of the star on the main sequence between points 1 and 2.

See: p448-449

In [45]:
total_ms = 92.9357

Myr = [
    94.4591, # 3
    94.5735, # 4
    94.9218, # 5
    95.2108,
    99.3835, # 7
    100.888, # 8
    107.208, # 9
    108.454    
]

print('stage points | elapsed time (Myr) | % of main sequence time')

for i in range(len(Myr)):
    
    n_1 = i + 2
    n_2 = i + 1 + 2
    
    if i > 0:
        elapsed = Myr[i] - Myr[i-1]
    else:
        elapsed = Myr[i] - total_ms
    
    percent = elapsed / total_ms * 100
    print(n_1, n_2, elapsed, percent)
    

stage points | elapsed time (Myr) | % of main sequence time
2 3 1.5234000000000094 1.6391978540001413
3 4 0.11439999999998918 0.12309586090166554
4 5 0.34830000000000894 0.3747752478326509
5 6 0.2890000000000015 0.3109676905645532
6 7 4.172699999999992 4.4898784858778615
7 8 1.5045000000000073 1.6188612126448794
8 9 6.319999999999993 6.800400707155585
9 10 1.2459999999999951 1.3407119115689612


**(b) How long does it take a 5$M_{\odot}$ star to cross the Hertzsprung gap relative to its main-sequence lifetime?**

Occurs between points `3-5`, so:

In [49]:
t = Myr[2] - Myr[0]
t / total_ms * 100, '%'

(0.4978711087343164, '%')

**(c) How long does the 5$M_{\odot}$ star spend on the blueward portion of the horizontal branch relative to its main-sequence lifetime?**

Ocurrs between `7-8`, so:

In [50]:
t = Myr[5] - Myr[4]
t / total_ms * 100, '%'

(1.6188612126448794, '%')

**(d) How long does the 5$M_{\odot}$ star spend on the redward portion of the horizontal branch relative to its main-sequence lifetime?**

Ocurrs  between `8-9`, so:

In [53]:
t = Myr[6] - Myr[5]
t / total_ms * 100, '%'

(6.800400707155585, '%')

# 13.6

In an attempt to identify the important components of AGB mass loss, various researchers have proposed parameterizations of the mass loss rate that are based on fitting observed rates for a specified set of stars with some general equation that includes measurable quantities associated with the stars in the sample. 

One of the most popular, developed by D. Reimers, is given by:

$$
M = -4*10^{-13}\eta \frac{L}{gR} M_{\odot} yr^{-1}
$$

(13.13)

where L, g, and R are the luminosity, surface gravity, and radius of the star, respectively (all in solar units; $g_{\odot} = 274 m s^{-2}$). $\eta$ is a free parameter whose value is expected to be near unity.

Note that the minus sign has been explicitly included here, indicating that the mass of the star is decreasing.

**(a) Explain qualitatively why L, g, and R enter Eq. (13.13) in the way they do.**



**(b) Estimate the mass loss rate of a 1$M_{\odot}$ AGB star that has a luminosity of 7000$L_{\odot}$ and a temperature of 3000 K.**

$L = 7000W$

$T = 3000K$

Assume $\eta = 1$

Need radius, R:

$$
R = \sqrt{\frac{L}{4 \pi \sigma T^4}} = 2.158*10^{11}m = 310R_{\odot}
$$

Surface gravity, $g$:

$$
g = G \frac M R^2 = 0.00285m/s-1
$$

$$
\frac g g_{\odot} = \frac{0.00285m/s-1}{274m/s-1} = 1.04*10^{-5}g_{\odot}
$$

$$
M = -4*10^{-13}\eta \frac{L}{gR} M_{\odot} yr^{-1} = -8.68*10^{-7} M_{\odot} yr^{-1}
$$

# 13.10
**(a) Using data available in Tables 12.1 and 13.1, compare the pre-main sequence evolutionary time of a 0.8$M_{\odot}$ star with the lifetime on the main sequence for a 15 $M_{\odot}$ star.**

How does this information help to explain the appearance of a color-magnitude diagram such as Fig. 13.18?

---

Massive stars moving off main sequence before low mass stars are still contracting in pre-main-sequence evolution and have not yet entered the main sequence.

**(b) Estimate the mass of a star that would have a main-sequence lifetime comparable to the
pre-main-sequence evolutionary time of a 0.8$M_{\odot}$ star.**

Answer:

Table 12.1 gives pre-main-sequence evolutionary time of a 0.8$M_{\odot}$ star as 68.5Myr.

This equates to the main sequence lifetime of approx 5-7$M_{\odot} star (Table 13.1).

# 13.11
**(a) The age of the universe is 13.7 Gyr. Compare this value to the main-sequence lifetime of a 0.8$M_{\odot}$ star. Why isn't it useful to compute the detailed post-main-sequence evolution of stars with masses much lower than the mass of the Sun?**

Main sequence lifetime of a 0.8$M_{\odot}$ star is 18.8 Gyr (Table 13.1, p 449), _which is older than the universe (13.7Gyr)._

Hard to compare theorectical models with obs - we don't have any yet!

**(b) Would you expect to find globular clusters with main-sequence turn-off points below 0.8$M_{\odot}$? Explain your answer.**

No - the universe isn't old enough (turn off points limited to age of universe).